deploy-tool/cert/cert.go

```diff
@@ -14,12 +14,17 @@ import (
 	"github.com/caddyserver/certmagic"
 )
 
-// IssueCertificate asks the Let's Encrypt ACMEApiEndpoint to validate and sign a certificate for a site using the DNS01 strategy
-func IssueCertificate(deployConfig config.DeployConfig, site string, privKey *ecdsa.PrivateKey) (string, error) {
-	dnsProvider, err := dns.GetACMEDNSProvider(deployConfig)
+// CertificateManager is a structure that defines the Certificate Issuance and renewal implementation
+type CertificateManager struct {
+	Config config.DeployConfig
+}
+
+func (certManager *CertificateManager) getCertMagic() (*certmagic.Config, error) {
```

Comment:
Go convention would be to name the receiver `cm` or even just `c`, which then makes it more obvious in the method body that you're referring to "this/self"